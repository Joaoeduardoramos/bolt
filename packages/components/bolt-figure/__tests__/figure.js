import {
  render,
  renderString,
  stop as stopTwigRenderer,
} from '@bolt/twig-renderer';
const { media } = require('./figure.data');

async function renderTwig(template, data) {
  return await render(template, data, true);
}

async function renderTwigString(template, data) {
  return await renderString(template, data, true);
}

describe('figure', async () => {
  afterAll(async () => {
    await stopTwigRenderer();
  });

  Object.keys(media).forEach(async item => {
    test(`figure with ${item}`, async () => {
      const results = await renderTwig('@bolt-components-figure/figure.twig', {
        media: {
          [item]: media[item],
        },
        caption: `Figure with ${item}.`,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });
});
