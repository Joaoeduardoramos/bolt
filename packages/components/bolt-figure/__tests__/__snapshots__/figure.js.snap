// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`figure figure with icon 1`] = `

<bolt-figure>
  <figure class="c-bolt-figure">
    <div class="c-bolt-figure__media">
      <bolt-icon name="add-open"
                 size="large"
      >
      </bolt-icon>
    </div>
    <figcaption class="c-bolt-figure__caption">
      Figure with icon.
    </figcaption>
  </figure>
</bolt-figure>

`;

exports[`figure figure with image 1`] = `

<bolt-figure>
  <figure class="c-bolt-figure">
    <div class="c-bolt-figure__media">
      <bolt-image style="background-color: hsl(233, 33%, 97%);">
        <bolt-ratio aspect-ratio-height="638"
                    aspect-ratio-width="1151"
                    style="--aspect-ratio-height: 638; --aspect-ratio-width: 1151;"
        >
          <img class="c-bolt-image__image-placeholder"
               sizes="auto"
               src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
          >
          <img class="c-bolt-image__image"
               srcset="/images/placeholders/landscape-16x9-mountains.jpg"
               sizes="auto"
               src="/images/placeholders/landscape-16x9-mountains.jpg"
          >
        </bolt-ratio>
      </bolt-image>
    </div>
    <figcaption class="c-bolt-figure__caption">
      Figure with image.
    </figcaption>
  </figure>
</bolt-figure>

`;

exports[`figure figure with table 1`] = `

<bolt-figure>
  <figure class="c-bolt-figure">
    <div class="c-bolt-figure__media">
      <bolt-table format="regular">
        <table class="c-bolt-table">
          <thead class="c-bolt-table__head">
            <tr class="c-bolt-table__row">
              <td class="c-bolt-table__cell">
              </td>
              <th class="c-bolt-table__cell c-bolt-table__cell--header"
                  scope="col"
              >
                Column 1
              </th>
              <th class="c-bolt-table__cell c-bolt-table__cell--header"
                  scope="col"
              >
                Column 2
              </th>
              <th class="c-bolt-table__cell c-bolt-table__cell--header"
                  scope="col"
              >
                Column 3
              </th>
            </tr>
          </thead>
          <tbody class="c-bolt-table__body">
            <tr class="c-bolt-table__row">
              <th class="c-bolt-table__cell c-bolt-table__cell--header"
                  scope="row"
              >
                Row 1
              </th>
              <td class="c-bolt-table__cell c-bolt-table__cell--data">
                R1C1
              </td>
              <td class="c-bolt-table__cell c-bolt-table__cell--data">
                R1C2
              </td>
              <td class="c-bolt-table__cell c-bolt-table__cell--data">
                R1C3
              </td>
            </tr>
            <tr class="c-bolt-table__row">
              <th class="c-bolt-table__cell c-bolt-table__cell--header"
                  scope="row"
              >
                Row 2
              </th>
              <td class="c-bolt-table__cell c-bolt-table__cell--data">
                R2C1
              </td>
              <td class="c-bolt-table__cell c-bolt-table__cell--data">
                R2C2
              </td>
              <td class="c-bolt-table__cell c-bolt-table__cell--data">
                R2C3
              </td>
            </tr>
            <tr class="c-bolt-table__row">
              <th class="c-bolt-table__cell c-bolt-table__cell--header"
                  scope="row"
              >
                Row 3
              </th>
              <td class="c-bolt-table__cell c-bolt-table__cell--data">
                R3C1
              </td>
              <td class="c-bolt-table__cell c-bolt-table__cell--data">
                R3C2
              </td>
              <td class="c-bolt-table__cell c-bolt-table__cell--data">
                R3C3
              </td>
            </tr>
          </tbody>
          <tfoot class="c-bolt-table__foot">
            <tr class="c-bolt-table__row">
              <th class="c-bolt-table__cell c-bolt-table__cell--header">
                Footer
              </th>
              <td class="c-bolt-table__cell c-bolt-table__cell--data">
                FC1
              </td>
              <td class="c-bolt-table__cell c-bolt-table__cell--data">
                FC2
              </td>
              <td class="c-bolt-table__cell c-bolt-table__cell--data">
                FC3
              </td>
            </tr>
          </tfoot>
        </table>
      </bolt-table>
    </div>
    <figcaption class="c-bolt-figure__caption">
      Figure with table.
    </figcaption>
  </figure>
</bolt-figure>

`;

exports[`figure figure with video 1`] = `

<bolt-figure>
  <figure class="c-bolt-figure">
    <div class="c-bolt-figure__media">
      <bolt-ratio aspect-ratio-height="9"
                  aspect-ratio-width="16"
                  style="--aspect-ratio-height: 9; --aspect-ratio-width: 16;"
      >
        <bolt-video class="js-bolt-video-uuid--12345"
                    data-setup="{&quot;techOrder&quot;: [&quot;Html5&quot;], &quot;resizeManager&quot;: false}"
                    video-id="3861325118001"
                    account-id="1900410236"
                    show-meta
                    show-meta-title
                    player-id="r1CAdLzTW"
                    controls
                    share-description="Share This Video"
        >
        </bolt-video>
      </bolt-ratio>
    </div>
    <figcaption class="c-bolt-figure__caption">
      Figure with video.
    </figcaption>
  </figure>
</bolt-figure>

`;
